Flink : UT testPrimaryKeyEqualToPartitionKey fails probablistcally

```
Test testPrimaryKeyFieldsAtEndOfTableSchema[catalogName=testhadoop, baseNamespace=default, format=PARQUET, isStreaming=true](org.apache.iceberg.flink.TestFlinkUpsert) failed with:
java.lang.AssertionError: 
Expecting:
  [+I[3, bbb, 2022-03-01], +I[1, aaa, 2022-03-01]]
to contain exactly in any order:
  [+I[2, aaa, 2022-03-01], +I[3, bbb, 2022-03-01]]
elements not found:
  [+I[2, aaa, 2022-03-01]]
and elements not expected:
  [+I[1, aaa, 2022-03-01]]

	at org.apache.iceberg.flink.TestHelpers.assertRows(TestHelpers.java:137)
	at org.apache.iceberg.flink.TestFlinkUpsert.testPrimaryKeyFieldsAtEndOfTableSchema(TestFlinkUpsert.java:262)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.apache.flink.util.TestNameProvider$1.evaluate(TestNameProvider.java:45)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
```



---

Looks like overwrite is not working properly and hence causing flakyness of this UT

produced locally with my set-up  also sample github run with failure : https://github.com/apache/iceberg/runs/5846777166?check_suite_focus=true

Another case failed in `testPrimaryKeyFieldsAtBeginningOfSchema`

```
org.apache.iceberg.flink.TestFlinkUpsert > testPrimaryKeyFieldsAtBeginningOfSchema[catalogName=testhadoop, baseNamespace=default, format=PARQUET, isStreaming=true] FAILED
    java.lang.AssertionError: 
    Expecting:
      [+I[bbb, 2022-03-01, 3], +I[aaa, 2022-03-01, 1]]
    to contain exactly in any order:
      [+I[aaa, 2022-03-01, 2], +I[bbb, 2022-03-01, 3]]
    elements not found:
      [+I[aaa, 2022-03-01, 2]]
    and elements not expected:
      [+I[aaa, 2022-03-01, 1]]
        at org.apache.iceberg.flink.TestHelpers.assertRows(TestHelpers.java:137)
        at org.apache.iceberg.flink.TestFlinkUpsert.testPrimaryKeyFieldsAtBeginningOfSchema(TestFlinkUpsert.java:[219](https://github.com/apache/iceberg/runs/5834902185?check_suite_focus=true#step:6:219))
```

--- a/flink/v1.13/flink/src/test/java/org/apache/iceberg/flink/TestFlinkUpsert.java
+++ b/flink/v1.13/flink/src/test/java/org/apache/iceberg/flink/TestFlinkUpsert.java
@@ -130,15 +130,15 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
 
     try {
       sql("INSERT INTO %s VALUES " +
-          "(1, 'a', TO_DATE('2022-03-01'))," +
-          "(2, 'b', TO_DATE('2022-03-01'))," +
-          "(1, 'b', TO_DATE('2022-03-01'))",
+          "(1, 'a', DATE '2022-03-01')," +
+          "(2, 'b', DATE '2022-03-01')," +
+          "(1, 'b', DATE '2022-03-01')",
           tableName);
 
       sql("INSERT INTO %s VALUES " +
-          "(4, 'a', TO_DATE('2022-03-02'))," +
-          "(5, 'b', TO_DATE('2022-03-02'))," +
-          "(1, 'b', TO_DATE('2022-03-02'))",
+          "(4, 'a', DATE '2022-03-02')," +
+          "(5, 'b', DATE '2022-03-02')," +
+          "(1, 'b', DATE '2022-03-02')",
           tableName);
 
       List<Row> rowsOn20220301 = Lists.newArrayList(Row.of(2, "b", dt20220301), Row.of(1, "a", dt20220301));
@@ -211,9 +211,9 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           tableName, toWithClause(tableUpsertProps));
 
       sql("INSERT INTO %s VALUES " +
-          "('aaa', TO_DATE('2022-03-01'), 1)," +
-          "('aaa', TO_DATE('2022-03-01'), 2)," +
-          "('bbb', TO_DATE('2022-03-01'), 3)",
+          "('aaa', DATE '2022-03-01', 1)," +
+          "('aaa', DATE '2022-03-01', 2)," +
+          "('bbb', DATE '2022-03-01', 3)",
           tableName);
 
       TestHelpers.assertRows(
@@ -221,8 +221,8 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           Lists.newArrayList(Row.of("aaa", dt, 2), Row.of("bbb", dt, 3)));
 
       sql("INSERT INTO %s VALUES " +
-          "('aaa', TO_DATE('2022-03-01'), 4)," +
-          "('bbb', TO_DATE('2022-03-01'), 5)",
+          "('aaa', DATE '2022-03-01', 4)," +
+          "('bbb', DATE '2022-03-01', 5)",
           tableName);
 
       TestHelpers.assertRows(
@@ -230,8 +230,8 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           Lists.newArrayList(Row.of("aaa", dt, 4), Row.of("bbb", dt, 5)));
 
       sql("INSERT INTO %s VALUES " +
-          "('aaa', TO_DATE('2022-03-01'), 6)," +
-          "('bbb', TO_DATE('2022-03-01'), 7)",
+          "('aaa', DATE '2022-03-01', 6)," +
+          "('bbb', DATE '2022-03-01', 7)",
           tableName);
 
       TestHelpers.assertRows(
@@ -254,9 +254,9 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           tableName, toWithClause(tableUpsertProps));
 
       sql("INSERT INTO %s VALUES " +
-          "(1, 'aaa', TO_DATE('2022-03-01'))," +
-          "(2, 'aaa', TO_DATE('2022-03-01'))," +
-          "(3, 'bbb', TO_DATE('2022-03-01'))",
+          "(1, 'aaa', DATE '2022-03-01')," +
+          "(2, 'aaa', DATE '2022-03-01')," +
+          "(3, 'bbb', DATE '2022-03-01')",
           tableName);
 
       TestHelpers.assertRows(
@@ -264,8 +264,8 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           Lists.newArrayList(Row.of(2, "aaa", dt), Row.of(3, "bbb", dt)));
 
       sql("INSERT INTO %s VALUES " +
-          "(4, 'aaa', TO_DATE('2022-03-01'))," +
-          "(5, 'bbb', TO_DATE('2022-03-01'))",
+          "(4, 'aaa', DATE '2022-03-01')," +
+          "(5, 'bbb', DATE '2022-03-01')",
           tableName);
 
       TestHelpers.assertRows(
@@ -273,8 +273,8 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           Lists.newArrayList(Row.of(4, "aaa", dt), Row.of(5, "bbb", dt)));
 
       sql("INSERT INTO %s VALUES " +
-          "(6, 'aaa', TO_DATE('2022-03-01'))," +
-          "(7, 'bbb', TO_DATE('2022-03-01'))",
+          "(6, 'aaa', DATE '2022-03-01')," +
+          "(7, 'bbb', DATE '2022-03-01')",
           tableName);
 
       TestHelpers.assertRows(

--- a/flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestFlinkUpsert.java
+++ b/flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestFlinkUpsert.java
@@ -130,15 +130,15 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
 
     try {
       sql("INSERT INTO %s VALUES " +
-          "(1, 'a', TO_DATE('2022-03-01'))," +
-          "(2, 'b', TO_DATE('2022-03-01'))," +
-          "(1, 'b', TO_DATE('2022-03-01'))",
+          "(1, 'a', DATE '2022-03-01')," +
+          "(2, 'b', DATE '2022-03-01')," +
+          "(1, 'b', DATE '2022-03-01')",
           tableName);
 
       sql("INSERT INTO %s VALUES " +
-          "(4, 'a', TO_DATE('2022-03-02'))," +
-          "(5, 'b', TO_DATE('2022-03-02'))," +
-          "(1, 'b', TO_DATE('2022-03-02'))",
+          "(4, 'a', DATE '2022-03-02')," +
+          "(5, 'b', DATE '2022-03-02')," +
+          "(1, 'b', DATE '2022-03-02')",
           tableName);
 
       List<Row> rowsOn20220301 = Lists.newArrayList(Row.of(2, "b", dt20220301), Row.of(1, "a", dt20220301));
@@ -211,9 +211,9 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           tableName, toWithClause(tableUpsertProps));
 
       sql("INSERT INTO %s VALUES " +
-          "('aaa', TO_DATE('2022-03-01'), 1)," +
-          "('aaa', TO_DATE('2022-03-01'), 2)," +
-          "('bbb', TO_DATE('2022-03-01'), 3)",
+          "('aaa', DATE '2022-03-01', 1)," +
+          "('aaa', DATE '2022-03-01', 2)," +
+          "('bbb', DATE '2022-03-01', 3)",
           tableName);
 
       TestHelpers.assertRows(
@@ -221,8 +221,8 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           Lists.newArrayList(Row.of("aaa", dt, 2), Row.of("bbb", dt, 3)));
 
       sql("INSERT INTO %s VALUES " +
-          "('aaa', TO_DATE('2022-03-01'), 4)," +
-          "('bbb', TO_DATE('2022-03-01'), 5)",
+          "('aaa', DATE '2022-03-01', 4)," +
+          "('bbb', DATE '2022-03-01', 5)",
           tableName);
 
       TestHelpers.assertRows(
@@ -230,8 +230,8 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           Lists.newArrayList(Row.of("aaa", dt, 4), Row.of("bbb", dt, 5)));
 
       sql("INSERT INTO %s VALUES " +
-          "('aaa', TO_DATE('2022-03-01'), 6)," +
-          "('bbb', TO_DATE('2022-03-01'), 7)",
+          "('aaa', DATE '2022-03-01', 6)," +
+          "('bbb', DATE '2022-03-01', 7)",
           tableName);
 
       TestHelpers.assertRows(
@@ -254,9 +254,9 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           tableName, toWithClause(tableUpsertProps));
 
       sql("INSERT INTO %s VALUES " +
-          "(1, 'aaa', TO_DATE('2022-03-01'))," +
-          "(2, 'aaa', TO_DATE('2022-03-01'))," +
-          "(3, 'bbb', TO_DATE('2022-03-01'))",
+          "(1, 'aaa', DATE '2022-03-01')," +
+          "(2, 'aaa', DATE '2022-03-01')," +
+          "(3, 'bbb', DATE '2022-03-01')",
           tableName);
 
       TestHelpers.assertRows(
@@ -264,8 +264,8 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           Lists.newArrayList(Row.of(2, "aaa", dt), Row.of(3, "bbb", dt)));
 
       sql("INSERT INTO %s VALUES " +
-          "(4, 'aaa', TO_DATE('2022-03-01'))," +
-          "(5, 'bbb', TO_DATE('2022-03-01'))",
+          "(4, 'aaa', DATE '2022-03-01')," +
+          "(5, 'bbb', DATE '2022-03-01')",
           tableName);
 
       TestHelpers.assertRows(
@@ -273,8 +273,8 @@ public class TestFlinkUpsert extends FlinkCatalogTestBase {
           Lists.newArrayList(Row.of(4, "aaa", dt), Row.of(5, "bbb", dt)));
 
       sql("INSERT INTO %s VALUES " +
-          "(6, 'aaa', TO_DATE('2022-03-01'))," +
-          "(7, 'bbb', TO_DATE('2022-03-01'))",
+          "(6, 'aaa', DATE '2022-03-01')," +
+          "(7, 'bbb', DATE '2022-03-01')",
           tableName);
 
       TestHelpers.assertRows(

